### PR TITLE
Fix `int_in_range` once and for all (hopefully)

### PR DESCRIPTION
**Pull request by fitzgen**

### CODE REVIEW

File: src/unstructured.rs [hunk 1]
```diff
@@ -319,18 +319,24 @@ impl<'a> Unstructured<'a> {
             return Ok((start, 0));
         }
 
-        let delta = end - start;
-        debug_assert_ne!(delta, T::ZERO);
+        // From here on out we work with the unsigned representation. All of the
+        // operations performed below work out just as well whether or not `T`
+        // is a signed or unsigned integer.
+        let start = start.to_unsigned();
```
Comment (Manishearth) [inline src/unstructured.rs:325]: Won't this fail for negative numbers?

Comment (fitzgen) [reply src/unstructured.rs:325]: The `to_unsigned` conversion is implemented with `as` so this call will not fail for negative numbers. Happy to rename the method to `as_unsigned` or something, but also not sure what we would call the `Int::from_unsigned` method in that case.

Comment (fitzgen) [reply src/unstructured.rs:325]: (See also the new test and fuzz target for signed integers and `int_in_range` which exercises this case)

Comment (Manishearth) [reply src/unstructured.rs:325]: Ah, I didn't realize `as` had those semantics

**Review by Manishearth — APPROVED** [review]

File: src/unstructured.rs (hunk 1 above)
Comment (fitzgen) [reply src/unstructured.rs:325]: yeah it is just a bitcast for e.g. `u8` to `i8`

**Comment by fitzgen** [comment]
Thanks for review!

**Pull request merged**